src/scripts/node_detector_control.py

```diff
@@ -21,29 +22,35 @@ def toggle_combine(self, msg):
             self.combine))
 
     def remove_object_detector(self, msg):
-        self.ready = False
-        self.detectors[msg.name].remove()
-        self.detectors.pop(msg.name)  # TODO: locks/thread safety? Is one boolean enough, or do we need the threading module?
-        self.ready = True
+        # Store detector to remove for quicker Lock releasing
+        detector_to_remove = self.detectors[msg.name]
```

Comment:
```suggestion
        if not msg.name in self.detectors:
            return 
        detector_to_remove = self.detectors[msg.name]
```

Vältetään keyerrorit

Reply:
Hyvä huomio, tämä lisätty. Lisäksi lisäsin tarkistukset sekä remove_, että add_object_detectoreihin tyhjien stringien varalta. En tosin tiedä hoitaako rossin viestit tämän jo valmiiksi, ei taida?